server/container/util-java8/src/main/java/org/apache/james/util/CompletableFutureUtil.java

```diff
@@ -38,4 +40,25 @@
             .orElse(CompletableFuture.completedFuture(Stream.of()));
     }
 
+    public static <T> CompletableFuture<Stream<T>> performOnAll(CompletableFuture<Stream<T>> futurStream, Function<T, CompletableFuture<Void>> action) {
+        return flatMapOnAll(futurStream, elem -> keepValue(() -> action.apply(elem), elem));
```

Comment:
I will replace it with value. Do you say elem is not a good variable name because it's a abbreviation of element or will you considere element to be also a bad variable name ? If yes why ?